include/geometric_shapes/check_isometry.h

```diff
@@ -0,0 +1,143 @@
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Martin Pecka, Robert Haschke */
+
+#ifndef GEOMETRIC_SHAPES_CHECK_ISOMETRY_H
+#define GEOMETRIC_SHAPES_CHECK_ISOMETRY_H
+
+#include <algorithm>
+#include <assert.h>
+#include <iostream>
+#include <iomanip>
+#include <math.h>
+#include <Eigen/Core>
+#include <Eigen/Geometry>
+
+/** \brief This file provides functions and macros that can be used to verify that an Eigen::Isometry3d is really an
+ * isometry. Eigen itself doesn't do the checks because they're expensive. If the isometry object is constructed in
+ * a wrong way (e.g. from an AngleAxisd with non-unit axis), it can represent a non-isometry. But some methods in the
+ * Isometry3d class perform isometry-specific operations which return wrong results when called on a non-isometry. E.g.
+ * for isometries, .linear() and .rotation() should be the same, but for non-isometries, the result of .linear()
+ * contains also the scaling factor, whether .rotation() is only the rotation part.
+ *
+ * These checks are primarily meant to be performed only in debug mode (via the ASSERT_ISOMETRY macro), but you can call
+ * checkIsometry() even in release mode. This check should be mainly performed on transforms input by the user.
+ */
+namespace bodies
+{
+#ifndef CHECK_ISOMETRY_PRECISION
+/** \brief The default precision to which the transform has to correspond to an isometry. */
+#define CHECK_ISOMETRY_PRECISION Eigen::NumTraits<double>::dummy_precision()
+#endif
+
+namespace
+{
+/**
+ * \brief Return the number of decimal digits that should be printed to print all digits important for the given float
+ *        precision. E.g. when floatPrecision is 1e-12, printPrecision is 12 + 1 = 13.
+ * \param floatPrecision The float precision to which float numbers are considered.
+ * \return The desired number of decimal digits (after decimal dot). The result will always be at least 6.
+ * \note Used as a parameter for std::setprecision().
+ */
+inline int getPrintPrecision(const double floatPrecision)
+{
+  return std::max(6, static_cast<int>(std::abs(std::log10(floatPrecision))) + 1);
+}
+}  // namespace
+
```

Comment:
Do we really need this? Why can't we simply go with default (scientific) formatting as before?

Reply:
Solved, displaying difference from the expected values in scientific format.